The following patch has been discussed on #ffmpeg-devel on 2008-06-02:

03:14:18 <@astrange> swscale+mmx2+x86-64 was fixed a while ago
03:15:03 <@DonDiego> siretart: one less patch for you..
12:44:59 < siretart> DonDiego: however, do you know what 'some time ago' means?
                     a commit number or at least a month would help me to
                     determine if the snapshot I have in debian is affected
                     or not
12:45:35 <@DonDiego> or wait, let me look at the logs..
12:46:40 <@DonDiego> r26270 | diego | 2008-03-22 19:22:53 +0100 (Sat, 22 Mar 2008) | 3 lines

status: still needs investigation/backport of that revision and a testbuild on amd64.


Index: ffmpeg.20080206/libswscale/swscale.c
===================================================================
--- ffmpeg.20080206.orig/libswscale/swscale.c	2008-01-29 15:58:10.000000000 +0100
+++ ffmpeg.20080206/libswscale/swscale.c	2008-03-15 23:11:50.000000000 +0100
@@ -1357,7 +1357,7 @@
     return 0;
 }
 
-#ifdef COMPILE_MMX2
+#if defined (COMPILE_MMX2) && !defined(ARCH_X86_64)
 static void initMMX2HScaler(int dstW, int xInc, uint8_t *funnyCode, int16_t *filter, int32_t *filterPos, int numSplits)
 {
     uint8_t *fragmentA;
@@ -2300,7 +2300,7 @@
                    srcFilter->chrH, dstFilter->chrH, c->param);
 
 #define MAX_FUNNY_CODE_SIZE 10000
-#if defined(COMPILE_MMX2)
+#if defined (COMPILE_MMX2) && !defined(ARCH_X86_64)
 // can't downscale !!!
         if (c->canMMX2BeUsed && (flags & SWS_FAST_BILINEAR))
         {
